GoogleForCreators/web-stories-wp
# PPC: Cards for issues

## Context

More cards for the updated prepublish checklist. This should be the rest of them!

## Summary

Sorry for the chunky PR, it was getting redundant chopping these up for smaller PRs so i thought I'd just do the rest of them all at once.

## Relevant Technical Choices

- Made a helper component for the `thumbnailBackground` for thumbnail types that aren't pages that are using the `IconLayer` from `getDefinitionByType` to decrease repetition, it's called `LayerThumbnail` 
- Made a util in `checklist` called `getActiveThumbnails`, as I type this I'm regretting the usage of `active`, i'll update that tomorrow. This util limits the thumbnails rendered if there's more than 4 instances of it. 
- Updated 1 check - the `storyTooLittleText` reads like it should be checking the page, and its sister check of tooMuchText is looking at the page so I went ahead and updated that to be `pageTooLittleText`, adjusted the test. 
- Otherwise this is pretty straightfoward. I added the cards and card type and the existing highlight. 
- I tried to clean up duplicate constants as I went, feel free to note anything you see that I missed. 

## To-do

- Need @embarks to help find the thing I'm missing in the highlights for navigating pages with the offending elements. I've got the text element getting focused but if you're not already on the page with that text on it the page doesn't update. 
- #8070 for thumbnail fallback when a poster isn't available

## User-facing changes

None, behind a feature flag. 

## Testing Instructions

Turn on the new checklist experiment and check for these issues:  (remember the paneling is getting done separate, just cards are showing up)

- pageTooManyLinks
- textElementFontTooSmall
- tooMuchPageText
- tooLIttlePageTExt
- videoElementResolution
- imageElementResolution
- videoElementResolution
- videoElementMissingPoster
- storyPagesCount 
- storyLengthTitle
- pageBackgroundLowTextContrast
- videoElementMissingDescription
- videoElementMissingCaptions
- elementLinkTappableRegionTOoSmall
- imageElementMissingAlt
- storyMissingExerpt
- storyPosterAttached
Checks should all work just like in the former checklist (while working on them I used the existing checklist as a sanity check for what was or wasn't showing up in the new one). 

### QA

N/A, behind flag 




This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Partially addresses #7920
